src/js/api/index.ts

```diff
@@ -182,6 +177,38 @@ const Commit = {
 
 const Deployment = {
   fetch: (id: string) => getApi<ApiEntityResponse>(`/api/deployments/${id}`),
+  fetchBuildLog: (id: string): Promise<ApiResult<string>> => {
+    const path = `${host}/ci/deployments/${id}/trace`;
+    const accessToken = getAccessToken();
+    const requestOptions: RequestInit = {
+      credentials: 'same-origin',
+      headers: {
+        Authorization: accessToken ? `Bearer ${accessToken}` : undefined,
+      },
+    };
+
+    return fetch(path, requestOptions)
```

Comment:
I would prefer writing this kind of logic with `await`. Makes things easier to follow.